Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Controllers/WindowsMixedRealityXRSDKArticulatedHand.cs

```diff
@@ -0,0 +1,333 @@
+                            }
+
+                            HandMeshInfo handMeshInfo = new HandMeshInfo
+                            {
+                                vertices = handMeshVertices,
+                                normals = handMeshNormals,
+                                triangles = handMeshTriangleIndices,
+                                uvs = handMeshUVs,
+                                position = translation.ToUnityVector3(),
+                                rotation = rotation.ToUnityQuaternion()
+                            };
+
+                            CoreServices.InputSystem?.RaiseHandMeshUpdated(InputSource, ControllerHandedness, handMeshInfo);
+                        }
+                    }
+                }
+            }
+#endif // WINDOWS_UWP
+        }
+
+        private void UpdateIndexFingerData(MixedRealityInteractionMapping interactionMapping)
+        {
+            // Update the interaction data source
+            interactionMapping.PoseData = unityJointPoses[TrackedHandJoint.IndexTip];
+
+            // If our value changed raise it.
+            if (interactionMapping.Changed)
+            {
+                // Raise input system event if it's enabled
+                CoreServices.InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, interactionMapping.PoseData);
+            }
+        }
+
+        private TrackedHandJoint ConvertToTrackedHandJoint(HandFinger finger, int index)
+        {
+            switch (finger)
+            {
+                case HandFinger.Thumb: return (index == 0) ? TrackedHandJoint.Wrist : TrackedHandJoint.ThumbMetacarpalJoint + index - 1;
+                case HandFinger.Index: return TrackedHandJoint.IndexMetacarpal + index;
+                case HandFinger.Middle: return TrackedHandJoint.MiddleMetacarpal + index;
+                case HandFinger.Ring: return TrackedHandJoint.RingMetacarpal + index;
+                case HandFinger.Pinky: return TrackedHandJoint.PinkyMetacarpal + index;
+                default: return TrackedHandJoint.None;
+            }
+        }
+
+        #region IMixedRealityHand Implementation
+
+        /// <inheritdoc/>
+        public bool TryGetJoint(TrackedHandJoint joint, out MixedRealityPose pose) => unityJointPoses.TryGetValue(joint, out pose);
+
+        #endregion IMixedRealityHand Implementation
+
+        private readonly float CursorBeamBackwardTolerance = 0.5f;
+        private readonly float CursorBeamUpTolerance = 0.8f;
+
+        /// <inheritdoc/>
+        public override bool IsInPointingPose
+        {
+            get
+            {
+                bool valid = true;
+                MixedRealityPose palmJoint;
+                if (unityJointPoses.TryGetValue(TrackedHandJoint.Palm, out palmJoint))
+                {
+                    Vector3 palmNormal = palmJoint.Rotation * (-1 * Vector3.up);
```

Comment:
Is it possible to share some of this pointing pose logic between this and the legacy path? Or do you think that because we'll get rid of the legacy path in a future iteration (i.e. post 2019), and given the lack of churn to this area, that it's fine to have a dupl-define?

Reply:
I can refactor this out into an ArticulatedHandUtils. I think yeah, we'll eventually get rid of the legacy path, but it's possible we'll want to update this between now and then. Sharing the logic definitely makes sense.

Reply:
I think I want to merge this as-is, then open a followup PR today refactoring out the shared code.